### PR TITLE
Fixes for January 2021 Update and Unassigned Citizens in Vanilla Banners

### PR DESCRIPTION
**Pull request by the-m4a**
Ready for review, as the proper fixes for the Linux stuff is likely a large scale one to deal with Firaxis' "fancy" way of doing file imports

**What's here:**
- Fixed issues with the banners after the January update
  - At some point, values for BannerType in CQUI and in the unmodified game became mismatched.  The specific break had to do with the "other district" BannerType
  - Firaxis removed the function to get the City Banner tooltip, which also broke CQUI + Vanilla
  - The changes to the Diplomacy screen required changing the includes declaration at the top of our one file, and deal with the change to the icon object (global variable in the expansion2, but local in the exp1/vanilla version).
- Fixed the issue with the unassigned citizens tooltip in Vanilla (issue #236 )
- Fixed issue where the tile yields would not display on hover for cities that had no available tiles to purchase
- Added annotations to the vanilla City Banner XML, including re-arranging a few lines so they would match when comparing the two in a file diff.
 
I pushed this to the Steam Workshop, and am playing now to see what else might've been missed.

### CODE REVIEW

File: Assets/UI/WorldView/citybannermanager.xml [hunk 1]
```diff
@@ -158,23 +162,24 @@
     <Instance Name="OtherCityBanner">
         <ZoomAnchor ID="Anchor" ZoomOffsetNear="0,0,0" ZoomOffsetFar="0,0,-20">
             <Container Anchor="C,C" Offset="0,-20" Size="200,34">
-                <Button ID="CityBannerButton" Anchor="C,T" Size="Parent-20,34" NoStateChange="true">
-                    <Grid ID="CityBannerBackground" Anchor="C,T" Offset="1,0" Size="Parent,34" Color="255,255,255,150" SliceTextureSize="50,34" SliceCorner="25,17" Texture="Banner_BannerBacking">
-                        <Image ID="CityProductionProgress" Anchor="R,C" Offset="2,0" Size="14,32" TextureOffset="14,0" Speed="1" Texture="Banner_ProductionMeter">
-                            <TextureBar ID="CityProductionNextTurn" Anchor="L,C" Size="14,32" Direction="Up" Speed="1" Texture="Banner_ProductionMeter"/>
-                            <TextureBar ID="CityProductionMeter" Anchor="L,C" Size="14,32" Direction="Up" Speed="1" Texture="Banner_ProductionMeter"/>
+                <Button ID="CityBannerButton" Size="Parent-20,34" Anchor="C,T" NoStateChange="true">
```
Comment (the-m4a) [inline Assets/UI/WorldView/citybannermanager.xml:165]: Any changes in this file beyond this point are just to match it to the unmodified version, specifically for the comparison of the files when using a file diffing tool.  None of the values were changed, just rearranged.  For example, line 172 is a change to the order of the XML attributes (putting Offset first, etc.) so the two files match.

File: Assets/UI/WorldView/citybannermanager_CQUI.lua [hunk 2]
```diff
@@ -22,11 +22,6 @@ BASE_CQUI_CityBanner_UpdateRangeStrike = CityBanner.UpdateRangeStrike;
 local COLOR_CITY_GREEN           = UI.GetColorValueFromHexLiteral(0xFF4CE710);
 local COLOR_CITY_RED             = UI.GetColorValueFromHexLiteral(0xFF0101F5);
 local COLOR_CITY_YELLOW          = UI.GetColorValueFromHexLiteral(0xFF2DFFF8);
-local BANNERTYPE_CITY_CENTER     = 0;
```
Comment (the-m4a) [inline Assets/UI/WorldView/citybannermanager_CQUI.lua:25]: these are declared as global values in the unmodified firaxis code and can be consumed by us (in fact, had been already for the mountain tunnel and the qh..one)

**Comment by the-m4a** [comment]
Converted this to draft in order to address the shenanigans with the wildcard includes that firaxis does for the CityBanners and the DiplomacyDealView.  I think I know how to fix it properly (we shouldn't use the ReplaceUIScript with the files where Firaxis does their wildcard include), but I need to do more work on that.

**Comment by the-m4a** [comment]
@alexeyOnGitHub as much as I wouldn't mind just assuming I'm a programming legend and never have any mistakes and therefore do not need a review before I can push changes... I think the policy that used to apply to master didn't get copied over to the main branch :D

**Comment by alexeyOnGitHub** [comment]
indeed, thank you for bringing this up. I added the same branch protection rules to the new "main" branch.

**Comment by the-m4a** [comment]
I am going to abandon this PR, and create a new one from the branch where I'd been sorting out the issues for Linux.  It contains all of these changes, plus the "correct" changes for the Banners and DiploDealView.  The changes are more extensive, but so far they work.
I want to look at the changes necessary for the interaction with Sukritact's mod before I push that one up to the workshop, though.

**Pull request closed without merging**